binsync/ui/panel_tabs/globals_table.py

```diff
@@ -34,25 +33,28 @@ def data(self, index, role=Qt.DisplayRole):
             return None
         col = index.column()
         row = index.row()
+        val = self.row_data[row][col]
         if role == Qt.DisplayRole:
-            if col == 0:
-                return self.row_data[row][col][0]
-            elif col == 1 or col == 2:
-                return self.row_data[row][col]
-            elif col == 3:
-                return friendly_datetime(self.row_data[row][col])
+            if col == GlobalsTableView.COL_TYPE:
+                return val[0]
+            elif col == GlobalsTableView.COL_ADDR:
+                return hex(val) if val is not None else ""
+            elif col in (GlobalsTableView.COL_NAME, GlobalsTableView.COL_USER):
+                return val
+            elif col == GlobalsTableView.COL_DATE:
+                return friendly_datetime(val)
         elif role == self.SortRole:
-            if col == self.time_col and isinstance(self.row_data[row][col], datetime.datetime):
-                return time.mktime(self.row_data[row][col].timetuple())
-            return self.row_data[row][col]
+            if col == self.time_col and isinstance(val, datetime.datetime):
+                return time.mktime(val.timetuple())
+            return val
         elif role == Qt.BackgroundRole:
             return self.data_bgcolors[row]
         elif role == self.FilterRole:
-            #print(self.row_data)
-            #print(self.row_data[row][0] + " " + self.row_data[row][1] + " " + self.row_data[row][2])
-            return self.row_data[row][0] + " " + self.row_data[row][1] + " " + self.row_data[row][2]
+            return " ".join(val for col in (
```

Comment:
I believe the bug is here

Reply:
You were right, the `val for col in ...` did not make any sense since `val` did depend on `col` in the loop. For readability purposes, I blindly replaced occurrences of `self.row_data[row][col]` with `val` in the function without thinking too much :sweat_smile:

Reply:
Also, addresses needed to be cast to `str` to be concatenated (it wasn't a problem before because addresses were printed inside the name column as strings)